src/styles/main.scss

```diff
@@ -1,3 +1,78 @@
+$cell-size: 100px;
+$gap-space: 1px;
+$calendar-size: $cell-size * 7 + $gap-space * 6;
+$cell-color: #eee;
+$cell-hover-color: pink;
+$cell-border-color: black;
+
+html {
+  font-family: "Arial", "Helvetica", sans-serif;
+  font-size: 30px;
+}
+
 body {
   margin: 0;
 }
+
+* {
+  box-sizing: border-box;
+}
+
+.calendar {
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  &__month {
+    display: flex;
+    width: $calendar-size;
+    flex-wrap: wrap;
+    gap: $gap-space;
+  }
+
+  @for $index from 1 through 31 {
+
+    &__day:nth-child(#{$index}) {
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      width: $cell-size;
+      height: $cell-size;
+      background-color: $cell-color;
+      border: 1px solid $cell-border-color;
+    }
+
+    &__day:nth-child(#{$index})::before {
+      content: "#{$index}";
+    }
+
+    &__day:nth-child(#{$index}):hover {
+      cursor: pointer;
+      background-color: $cell-hover-color;
+      transform: translateY(-20px);
+      transition: 0.5s;
+    }
+  }
+
+  $days:
+    "mon" 0,
+    "tue" $cell-size + $gap-space,
+    "wed" ($cell-size + $gap-space) * 2,
+    "thu" ($cell-size + $gap-space) * 3,
+    "fri" ($cell-size + $gap-space) * 4,
+    "sat" ($cell-size + $gap-space) * 5,
+    "sun" ($cell-size + $gap-space) * 6;
+
+  @each $week-day, $value in $days {
+    &--start-day--#{$week-day} &__day:first-child {
+      margin-left: $value;
```

Comment:
if linter allows.
 You can write margin-left: ($cell-size + $gap-space)  * $value not to repeate your code.

Reply:
Agree